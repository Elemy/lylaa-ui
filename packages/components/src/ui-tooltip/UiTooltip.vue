<script setup lang="ts">
import { Tooltip } from "ant-design-vue";
import { usePrefixCls } from "../composables/useUtils";
</script>

<template>
  <component
    :is="Tooltip"
    v-bind="$attrs"
    :prefix-cls="usePrefixCls('tooltip')"
  >
    <slot></slot>
  </component>
</template>

<style lang="less"></style>
