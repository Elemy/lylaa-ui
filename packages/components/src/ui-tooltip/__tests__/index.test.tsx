import { describe, expect, it, vi } from "vitest";
import { mount } from "@vue/test-utils";
import UiTooltip from "../UiTooltip.vue";

// const waitRAF = () => new Promise((resolve) => requestAnimationFrame(resolve));
// beforeEach(() => {
//   vi.spyOn(window, "requestAnimationFrame").mockImplementation(
//     (cb: FrameRequestCallback): number => {
//       cb(0);
//       return 0;
//     }
//   );
// });

// afterEach(() => {
// window.requestAnimationFrame.mockRestore();
// });

describe("Button", () => {
  it("鼠标移入", async () => {
    const wrapper = mount({
      render() {
        return (
          <UiTooltip title="hello lyla ui">
            <div id="hello" class="hello-class">
              hello Lylaa ui
            </div>
          </UiTooltip>
        );
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    await wrapper.find("#hello").trigger("mouseenter");
    await new Promise((resolve) => setTimeout(resolve, 200));
    expect(wrapper.find("#hello").classes()).toContain("testui-tooltip-open");
  });
});
