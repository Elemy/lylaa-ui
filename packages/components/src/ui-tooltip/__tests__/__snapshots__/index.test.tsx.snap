// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > 鼠标移入 1`] = `"<div id=\\"hello\\" class=\\"hello-class\\">hello Lylaa ui</div>"`;
